batchflow/models/metrics/regression.py

```diff
@@ -0,0 +1,134 @@
+        Ground truth (correct) target values.
+
+    predictions : array-like of shape = (n_samples) or (n_samples, n_outputs) \
+                  or list of such arrays
+        Estimated target values
+
+    multi: bool
+        Whether the task is multiouput (default is False)
+
+    weights : array-like of shape (n_samples)
+        Sample weights.
+
+    gap : float
+        Max difference between target and prediction for sample to be \
+        considered as properly classified (default is 3)
+
+    Notes
+    -----
+    - Accuracy metric for regression task stands for the ratio of samples for which `abs(targets-predictions) < gap`.
+
+    - For all the metrics, except max error, accuracy and median absolute error, you can compute sample-wise weighting.
+    For that purpose specify `weight` argument, which must be the same size as inputs.
+
+    In the multioutput case metrics might be calculated with or without outputs averaging.
+
+    Available methods are:
+
+    - `None` - no averaging, calculate metrics for each output individually
+    - `mean` - calculate metrics for each output, and take their mean.
+
+    **Metrics**
+    All metrics return:
+
+    - a single value if targets are single output i.e. have shape (n_samples, ).
+    - a single value if targets are multioutput i.e have shape (n_samples, n_outputs) and `agg` set to `mean`
+    - a vector with (n_outputs, ) items if targets are multioutput and `agg` set to `None`
+    """
+
+    def __init__(self, targets, predictions, weights=None, multi=False):
+        super().__init__()
+
+        # if-else block bellow process the case when the inputs and targets are list
+        # of arrays-like. Thats happening when we accumulating targets and predictions in the pipeline
+        # via `.update(V('targets', mode='a'), value)`.
+        # Since we are not interested in across batch aggregation we only need to concatenate this arrays along 0 axis
+        # In this scenario tho we can mess with 2 cases:
+        # 1. targets is a single batch of multioutput targets and has the following structure [[], [], ..].
+        # 2. targets is a list of data come from couple of batches in a single output task
+        # and has the same structure [[], [], ..].
+        # In the first case concatenating such arrays do not have any sence. To separate such cases we introduce
+        # 'multi' argument.
+
+        if np.ndim(targets) == 1 or (np.ndim(targets) == 2 and multi):
+            self.targets = np.array(targets)
+            self.predictions = np.array(predictions)
+        else:
+            self.targets = np.concatenate(targets, axis=0)
+            self.predictions = np.concatenate(predictions, axis=0)
+
+        if weights is not None:
+            self.weights = np.array(weights).flatten()
+        else:
+            self.weights = None
+
+        self.multi = multi
+        self._agg_fn_dict = {
+            'mean': lambda x: np.mean(x),
+        }
```

Comment:
I would rather update existing parent class field, not re-write it
What about moving this initialization to base class?